src/app/frontend/replicasetlist/replicasetlistcontainer.js

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Computes optimal height of the given container that will fit all its child elements into N
+ * column layout. N is based on current media query from the $mdMedia service.
+ *
+ * @param {!Element} container Container element that has child elements to be arranged in columns.
+ *     It should be a flexbox container with columns wrapped.
+ * @param {function(string):boolean} mdMedia Angular Material $mdMedia service
+ * @return {number}
+ */
+export default function computeContainerHeight(container, mdMedia) {
+  /** @type {!Array<number>} */
+  let childHeights = Array.prototype.map.call(container.children, (child) => child.offsetHeight);
+
+  let columnCount = computeColumnCount(mdMedia);
+  return binarySearchOptimalHeight(childHeights, columnCount);
```

Comment:
This is a great non-trivial solution. I see that complexity here is `O(nlog(n))` so it's really good. I'm not sure how big sets of data will be used here, but for most cases this should be sufficient. 

I was thinking a bit about other possible solutions(I like algorithmic stuff :)). I think there are also solutions for this NP problem with complexity `O(n)`. As example, there could be used algorithm to split array of heights into P subarrays of balanced sum(`O(P*n)`). The subarray with highest partial sum would be our height.

This is very similar to bin packing problem.